code/game/objects/structures/barsign.dm

```diff
@@ -53,6 +53,15 @@
 	"thesyndicat",
 	"error"))
 
+/obj/structure/sign/double/barsign/carp
+	name = "The Drunk Carp"
+	desc = "The Drunk Carp, Bar and Grill"
+	icon_state = "thedrunkcarp"
+
+/obj/structure/sign/double/barsign/carp/Initialize()
+	. = ..()
+	ChangeSign(pick(
+	"thedrunkcarp"))
```

Comment:
Is there every more than one option?
Otherwise just remove this

Reply:
I did this since it kept randomizing the sign if I set it with icon_state. Very sloppy, but I don't know another way to do it.

Reply:
What i meant is you can keep the ChangeSign just remove the pick inside
Since your picking from a list of one you could just do `ChangeSign("thedrunkcarp")`